CHANGELOG.md

```diff
@@ -6,6 +6,10 @@
 
 #### New Features
 
+- Updated `Session` class to be thread-safe. This allows concurrent dataframe transformations, dataframe actions, UDF and stored procedure registration, and concurrent file uploads when using the same `Session` object.
+  - The feature is disabled by default and can be enabled by setting `FEATURE_THREAD_SAFE_PYTHON_SESSION` to `True` for account.
```

Comment:
This is server side parameter right? It's not visible to users. You might want to mention client side parameter here

Reply:
There is not client side param for this. The parameter if we use real locks or dummy locks which is done during the `__init__` phase. The other way to do it is using `session_parameters` when building the connection configs but 1. that is not documented properly, and 2. it will not enable the server-side changes when running inside stored proc. 

So I decided to keep it at account level.

Reply:
I see, thanks for explanation